dace/memlet.py

```diff
@@ -362,11 +362,21 @@ def try_initialize(self, sdfg: 'dace.sdfg.SDFG',
             # Cannot initialize yet
             return
 
-        is_data_src = True
+        is_data_src = False
+        is_data_dst = False
+        if isinstance(path[0].src, AccessNode):
+            if path[0].src.data == self._data:
+                is_data_src = True
         if isinstance(path[-1].dst, AccessNode):
             if path[-1].dst.data == self._data:
-                is_data_src = False
-        self._is_data_src = is_data_src
+                is_data_dst = True
+        if is_data_src and is_data_dst:
```

Comment:
Is this potentially also causing us issues, i.e., do we need to know with great confidence the data source?